pyflakes/checker.py

```diff
@@ -588,6 +587,17 @@ def _add_to_names(container):
 
 class Scope(dict):
     importStarred = False       # set to True when import * is found
+    # Special key for checking whether a binding is defined only for type checking.
+    TYPE_CHECKING_ONLY = object()
+
+    def __init__(self):
+        super(Scope, self).__init__(self)
+        self[self.TYPE_CHECKING_ONLY] = collections.defaultdict(bool)
```

Comment:
this is worse imo, this would be typed very strangely `Dict[Union[object, str], Union[Binding, Dict[str, bool]]` -- I think it would be better to have two stacks than this weirdness.  Also `Dict[str, bool]` is better represented as `Set[str]`